### PR TITLE
Flow Help Docs

### PR DESCRIPTION
**Pull request by waigani**
Add Flow and Decorator Help Templates to the Flow SDK to render help for each
Flow. Make each Flow construct a CLI and Decorator App in stead of Command.
This fixes a bug where the option flags were being displayed incorrectly.

### CODE REVIEW

File: flows/codelingo/review/Gopkg.lock [hunk 1]
```diff
@@ -35,7 +35,7 @@
   version = "1.3"
```
Comment (emersonwood) [inline flows/codelingo/review/Gopkg.lock]: I know this PR didn't introduce this, but I'm wondering why we have a nested Gopkg.lock

Comment (emersonwood) [inline flows/codelingo/review/Gopkg.lock]: I know this PR didn't introduce this, but I'm wondering why we have a nested Gopkg.lock

File: flows/codelingo/review/Gopkg.toml [hunk 2]
```diff
@@ -30,7 +30,7 @@
   version = "1.3.0"
```
Comment (emersonwood) [inline flows/codelingo/review/Gopkg.toml]: And nested Gopkg.toml

File: sdk/flow/cli.go [hunk 3]
```diff
@@ -25,45 +27,119 @@ type DecoratedResult struct {
...
+
+	*cliApp = app
+}
+
+func (f *flowRunner) Run() (_ chan *DecoratedResult, err error) {
+	return f.decoratedResultc, f.cliApp.Run(os.Args)
+}
+
+func (f *flowRunner) action(ctx *cli.Context) {
+	panic("action run")
```
Comment (emersonwood) [inline sdk/flow/cli.go]: debug code?

File: flows/codelingo/review/Gopkg.lock (hunk 1 above)
Comment (waigani) [reply flows/codelingo/review/Gopkg.lock]: we use to vendor the flows individually. Good catch, I'll remove.

**Pull request merged**